File: Examples/VisualBasic/Working-with-External-Data-Sources/EditDataConAndRefreshRecords.vb
Imports Aspose.Diagram
Imports System
Public Class EditDataConAndRefreshRecords
    Public Shared Sub Run()
        ' ExStart:EditDataConAndRefreshRecords
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_ExternalDataSources()

        ' Load source Visio diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsd"))
        ' Set connecting string
        diagram.DataConnections(0).ConnectionString = "Data Source=MyServer;Initial Catalog=MyDB;Integrated Security=True"
        ' Set command
        diagram.DataConnections(0).Command = "SELECT * from Project with(nolock)"
        ' Refresh all record sets
        diagram.Refresh()
        ' Save Visio diagram
        diagram.Save(dataDir & Convert.ToString("EditDataConAndRefreshRecords_out.vdx"), SaveFileFormat.VDX)
        ' ExEnd:EditDataConAndRefreshRecords
    End Sub
End Class

File: Examples/CSharp/Working-with-External-Data-Sources/EditDataConAndRefreshRecords.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_with_External_Data_Sources
{
    public class EditDataConAndRefreshRecords
    {
        public static void Run()
        {
            // ExStart:EditDataConAndRefreshRecords
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ExternalDataSources();

            // Load source Visio diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsd");
            // Set connecting string
            diagram.DataConnections[0].ConnectionString = "Data Source=MyServer;Initial Catalog=MyDB;Integrated Security=True";
            // Set command
            diagram.DataConnections[0].Command = "SELECT * from Project with(nolock)";
            // Refresh all record sets
            diagram.Refresh();
            // Save Visio diagram
            diagram.Save(dataDir + "EditDataConAndRefreshRecords_out.vdx", SaveFileFormat.VDX);
            // ExEnd:EditDataConAndRefreshRecords
        }
    }
}
